graph/graph.go

```diff
@@ -235,6 +235,11 @@ func (graph *Graph) Register(img *image.Image, layerData archive.Reader) (err er
 	graph.imageMutex.Lock(img.ID)
 	defer graph.imageMutex.Unlock(img.ID)
 
+	// Skip register if image is already registered
+	if graph.Exists(img.ID) {
```

Comment:
This changes the contract of the method, somewhat. Before, one could get an error and know that they need to close the `archive.Reader`. Now, they have now way of knowing that is has been consumed.

Either way, this method *should* called `layerData.Close` to maintain the contract of the `archive.Reader`, which is really just a `io.ReadCloser`.